client/src/actions/index.js

```diff
@@ -27,21 +28,43 @@ export const signOut = () => {
 
 export const createStream = formValues => async (dispatch, getState) => {
   const { userId } = getState().auth;
-  const payload = {
-    description: formValues.description,
-    title: formValues.title,
-    userid: userId
-  };
+  let streamLink;
   database
-    .collection("streams")
-    .doc(userId)
-    .set(payload)
-    .then(() => {
-      dispatch({ type: CREATE_STREAM, payload: payload });
-      history.push("/");
+    .collection("streamlinks")
+    .get()
+    .then(querySnapshot => {
+      const data = querySnapshot.docs.map(doc => doc.data());
+
+      for (var i = 0; i < data.length; i++) {
+        if (!data[i].used) {
+          streamLink = data[i].link;
+          break;
+        }
+      }
+      if (streamLink == null) {
+        console.log("No more stream links");
```

Comment:
I think instead of this, we should just not show the create stream button to the user. I'll create a ticket.